### DISCUSSION TITLE
FreeCAD to OpenEMS Plugin

### DISCUSSION THREAD

**Discussion by Romi975** [LubomirJagos/FreeCAD-OpenEMS-Export#31] [Q&A]
Hi, I need help with this Plugin for the Port Settings. I cant see the option of nf2ff and ht dump and also in the Probe Settings there is no Settings for nf2ff box and ht Dump available. Can you kindly upload the older version of the plugin? Thanks

**Comment by LubomirJagos42** [comment]
hi, I removed that older versions due there were errors in generated octave code and I moved EH dumps into probes and therefore made changes for .ini file

like E or H dumps are in probes and have options there, also nf2ff box has settings in processing tab where you can then generate script for it

Can you post here picture what you are missing? I tested dumps and they were working

**Reply by Romi975** [reply]
![github ](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/ed3a009d-6a68-4e8f-8310-5762f153e3d1)
This is confusing me because in my design of planar Antenna the far field port should have nf2ff and a H dump type but now there is curve and three other which are new for me.

**Reply by LubomirJagos** [reply]
I moved nf2ff box, E and H field dumps into probe tab as seen on picture below, due first when I learned openEMS their functions in octave code are under addPort() but in real simulation they are inactive ports for dumping data, so they are probes, which makes more sense to me as they are probing fields, also it's similar to HFSS and CST and other EM simulation software, so you have to specify them on probe tab
then for nf2ff box you can set params in processing tab and choose one of nf2ff box and generate script to calculate far field
![image](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/18229825/0eaf7452-6441-4d23-bbbe-cbdfdf398451)
![image](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/18229825/23598217-20c2-4f84-94b6-072ee84dda99)

**Reply by LubomirJagos** [reply]
also since old versions I repaired lots of errors in generated octave code, they are not obvious in simple models but there were some and reason why are created separate category for probe is also there are priorities specified for ports but they have no influence on model for dumpboxes and therefore all probes has priority set to 0
I can like put there old releases but I would like to not since there are errors in generated codes, you can find them in git history, like you can clone whole repository and revert it to some commits, but I would recommend use the release, I still correcting code generator, nowaday mostly to generate right code for polar coordinates since that part was never tested

don't worry too much if generated code is right or not, most important is to check your model in CSXCAD viewer, that's most important check what you have to do, I always generated script and then spend lot of time looking on it to, mostly on mesh and ports if there are intersections between them right

**Reply by Romi975** [reply]
I have tried choosing the NF2FF box but still no results. The CSDCAD is working even with Probe Type -> Probe but theres no data. This is the example how the Graphs should look like 
![Diagramme](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/5d4f1985-f724-4bfe-8bd9-1a811bef8f61) this in my case these are like->
![diagramme 2](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/359958a9-7355-436c-9e86-a952fb0aef8f)
![diagramm 3](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/bf7e8460-3f07-4170-9586-0230afdcb39e) . This is the pic of CSXCAD of my Antenna model

![AppCSXCAD](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/d75ca713-4a9d-4431-91ee-692bc903993d)

**Comment by Romi975** [comment]
I have attached my files in the zip folder. Please if you can spare some time to look inside and see if there are any issues with my FreeCAD model or the plugin settings. 
[Antenna.zip](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/files/13342657/Antenna.zip)

**Reply by LubomirJagos** [reply]
hi, just briefly looking on your model there are these mistakes which are related to your model:
    - you are using port 1 as feed port but there are no gridlines in XY crossing it, use fixed count and set it to 3 or 4 at least in X and Y direction to be sure you have som gridlines intersection inside your port
    - your substrate are around your patch is too small, increase ground and substrate dimensions to have at least lambda/4 around your patch, now your patch antenna is more like capacitor so even if port is set right it's not patch antenna
    
Unluckily I don't have model of patch now at PC, I can create one, but now I need to finished work for my job, so hopefully these recommendation will be enough to make it running, I can create one correct it's not hard.

**Reply by LubomirJagos** [reply]
I modified your script by adding gridlines into port due you are using spacing 1.4mm but port is smaller than that and also made luft grid more dense to have space for 8 PML cells on boundaries, but still you need to increase area of substrate and ground to make it patch instead capacitor
![image](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/18229825/67552b7a-a195-4622-b36a-4fe6289e3c29)

modified simulation and S11 analysis script:
[raheemoctave_openEMS_simulation.zip](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/files/13343534/raheemoctave_openEMS_simulation.zip)

**Reply by Romi975** [reply]
yes that graph looks pretty good. I have download this file but now the issue is that output simulation is missing. Kindly could you upload the modified patch antenna model with plugin settings what you have made?

**Reply by LubomirJagos** [reply]
Hi,
saw I played with patch antenna simulation like you want to do and here is my test file, but I  tried to do what you are trying to do and there are reasons why it's not running as you expecting and it's wrong way, here there are:
    - excitation port placed directly on patch - you never know what is input impedance and to make this running right you have to place your excitation port on EXACT place, patch has only 1 resonant frequency and that place is too sharp, E field when is misaligned starts to rotate and energy is not decreasing, excitation point must be on patch axis somewhere for patch to perform right
    - you are missing feed point transmission line which transform 50Ohm to patch input impedance, it's microstrip input feedline usually lambda/4 transformer which transforms 50Ohm to patch input impedance, there are calculator on internet how to calculate it

Attaching my file which is good start point, but as I mentioned this model is really trying to be TOO SHARP and RESULT is BAD, you need to add input microstrip transformer and then thing starts to turn right. I will try to make that simulation as I need to add it to examples, but this one is too hard to get it right and it's just academic one, since feedpoint is directly on patch.
![image](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/18229825/511efdcc-1814-4b2f-94b8-b7c3e7eabcb1)
![image](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/18229825/702d6e77-d9fc-440f-ada6-02a93de3c840)

[raheem patch antenna.zip](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/files/13401588/raheem.patch.antenna.zip)
[openEMS_simulation_s11_dB.csv](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/files/13401593/openEMS_simulation_s11_dB.csv)

**Comment by Romi975** [comment]
Thanks a  lot for your time. I have run your model and it has no errors so far, everything is running well.

**Comment by Romi975** [comment]
hi, could you please tell me how to get rid of this error? Its not showing the figure in Octave anymore. 
![j](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/assets/115326621/7c765319-71f1-4be1-9e90-7ba87200630f)

**Reply by LubomirJagos** [reply]
there are no simulation files in simulation_output folder, there should be file port_ut1 but probablz you run simulation more times and removed that folder or whatever, in simulation_output there outputs from simulation that one is text file where are values of voltage in time and there should be another port_it1 for current values over time

**Comment by Romi975** [comment]
is it possible to have your email address or contact?

**Reply by LubomirJagos** [reply]
sure, I have linkedin profile and there is my mail, also it's in every my commit in history visible
https://www.linkedin.com/in/lubomirjagos/
